### PR TITLE
✨ gcp: add Vertex AI data-movement & training job resources

### PR DESCRIPTION
**Pull request by tas50**
## What

Adds five Vertex AI job/pipeline resources under `gcp.project.vertexaiService`:

| Resource | Audit value |
|----------|-------------|
| `batchPredictionJob` | `inputConfig`/`outputConfig` expose the GCS/BigQuery data sources & destinations — **batch data-egress audit**; plus `model`, `serviceAccount`, `disableContainerLogging` |
| `tuningJob` | managed fine-tuning — `baseModel`, `tunedModel`, `tuningDataStats`, `serviceAccount` (where is training data going, what base model) |
| `trainingPipeline` | `inputDataConfig`, `trainingTaskDefinition`, model linkage, `state` |
| `hyperparameterTuningJob` | `studySpec`, `trialJobSpec`, trial-count bounds |
| `modelDeploymentMonitoringJob` | endpoint drift/skew monitoring — `endpoint`, `scheduleState`, `nextScheduleTime` |

## Why

These are the Vertex AI surfaces where data actually moves in and out of models and where training happens — high value for shadow-AI and data-governance audits, all GA, all on `aiplatform.googleapis.com`.

## Implementation

Each resource exposes `encryptionSpec` with a resolved `kmsKey()` reference, `labels`, and lifecycle timestamps. Listed across all Vertex AI regions via the existing `listAcrossRegions` fan-out using `JobClient`, `GenAiTuningClient`, and `PipelineClient`. New `.lr.versions` entries at `13.20.1`. Adds five `aiplatform.*.list` permissions.

## Example

```mql
// batch jobs reading/writing outside the project
gcp.project.vertexai.batchPredictionJobs {
  displayName
  state
  inputConfig
  outputConfig
  serviceAccount
}
```

## Testing

Generated (`./lr go` + `./lr versions`), built, and installed the gcp provider locally. No live-cloud verification in this PR (build + generate only, per scope).

### CODE REVIEW

File: providers/gcp/resources/gcp.lr [hunk 1]
```diff
@@ -11133,6 +11143,217 @@ private gcp.project.vertexaiService.indexEndpoint @defaults("displayName") {
...
+// reports lifecycle status. Encryption at rest is reported through
+// `encryptionSpec` and the resolved `kmsKey()`.
+private gcp.project.vertexaiService.trainingPipeline @defaults("displayName state") {
+  // Full resource name
+  name string
+  // Display name
+  displayName string
+  // Cloud Storage URI of the training task definition schema
+  trainingTaskDefinition string
+  // Input data configuration (dataset and data split)
```
[the remark this answers is not in the document]
Comment (tas50) [reply providers/gcp/resources/gcp.lr:11645]: Done — replaced `model string` with a computed `model() gcp.project.vertexaiService.model` typed ref. The raw resource name is cached in `cacheModelName` on the Internal struct and resolved via `NewResource` on demand.

File: providers/gcp/resources/gcp.lr [hunk 2]
```diff
@@ -11524,6 +11534,217 @@ private gcp.project.vertexaiService.cachedContent @defaults("displayName model")
...
+  // Full resource name
+  name string
+  // Display name
+  displayName string
+  // Current state (JOB_STATE_QUEUED, PENDING, RUNNING, SUCCEEDED, FAILED, CANCELLING, CANCELLED, PAUSED, EXPIRED, UPDATING, PARTIALLY_SUCCEEDED)
+  state string
+  // Maximum number of trials to run
+  maxTrialCount int
+  // Number of trials to run in parallel
+  parallelTrialCount int
```
[the remark this answers is not in the document]
Comment (tas50) [reply providers/gcp/resources/gcp.lr:11688]: Done — `parentModel` is now a typed `parentModel() gcp.project.vertexaiService.model` accessor (raw resource name cached on the Internal struct, resolved via `NewResource`).

I also went a step further on `modelId`: rather than keep it as a plain string, I made it a typed `model() gcp.project.vertexaiService.model` ref. The training pipeline is listed per-region, so we have the project + region context to construct the full model resource name (`projects/{project}/locations/{region}/models/{modelId}`) and resolve it — letting users traverse `trainingPipeline.model.displayName` directly.
